sd-finetuner-workflow/sd-finetuner/finetuner.py

```diff
@@ -452,34 +440,34 @@ def __init__(
         self.train_dataloader = train_dataloader
         self.noise_scheduler = noise_scheduler
         self.lr_scheduler = lr_scheduler
-        self.lr_scaler = lr_scaler
         self.optimizer = optimizer
-        self.device = device
         self.weight_dtype = weight_dtype
         self.args = args
 
-        self.progress_bar = tqdm.tqdm(
-            range(args.epochs * len(self.train_dataloader)),
-            desc="Total Steps",
-            leave=False,
-        )
-        self.run = wandb.init(
-            project=args.project_id,
-            name=args.run_name,
-            config={
-                k: v for k, v in vars(args).items() if k not in ["hf_token"]
-            },
-            dir=args.output_path + "/wandb",
-        )
-        self.global_step = 0
+        if accelerator.is_main_process:
+            self.progress_bar = tqdm.tqdm(
```

Comment:
`tqdm` is pretty terrible for Argo workflows and logging. Look at the finetuner for how I work with it.